src/org/opensolaris/opengrok/analysis/pascal/PascalSymbolTokenizer.lex

```diff
@@ -0,0 +1,83 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").  
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2006, 2012, Oracle and/or its affiliates. All rights reserved.
+ */
+
+/*
+ * Gets Pascal symbols - ignores comments, strings, keywords
+ */
+
+package org.opensolaris.opengrok.analysis.pascal;
+import org.opensolaris.opengrok.analysis.JFlexTokenizer;
+
+%%
+%public
+%class PascalSymbolTokenizer
+%extends JFlexTokenizer
+%init{
+super(in);
+%init}
+%unicode
+%type boolean
+%eofval{
+this.finalOffset =  zzEndRead;
+return false;
+%eofval}
+%char
+
+Identifier = [a-zA-Z_] [a-zA-Z0-9_]*
+
+%state STRING COMMENT SCOMMENT QSTRING
+
+%%
+
+<YYINITIAL> {
+{Identifier} {String id = yytext();
+                if(!Consts.kwd.contains(id)){
```

Comment:
alt. of course work with ctags devs and get them to support object pascal
they are pretty responsive over at universal ctags, so that might be doable, too